Core/Object Arts/Dolphin/IDE/Base/SmalltalkSearchPresenter.cls

```diff
@@ -154,14 +154,21 @@ rangeToCompleteAt: anInteger
 				ifFalse: [anInteger to: view caretPosition - 1]]!
 
 searchItemsStartingWith: aString maxItems: anInteger 
-	| result classes |
+	| classNamePrefix result classes |
 	aString first isLowerCase 
 		ifTrue: 
 			[| matchingSymbols |
 			matchingSymbols := self selectorsStartingWith: aString maxItems: anInteger.
 			^matchingSymbols collect: [:each | each displayString]].
+	classNamePrefix := aString upTo: $>.
```

Comment:
Good idea, but how about something like the following, which also delays including metaclasses. Without that the "classes size > 1" condition below is going to be true even when there is really only one matching class.

```
	filter := (aString includes: $>) 
				ifTrue: 
					[| classNamePrefix |
					classNamePrefix := aString upTo: $>.
					[:each | each name asString = classNamePrefix]]
				ifFalse: 
					[(aString includes: Character space) 
						ifTrue: [[:each | each name beginsWith: aString]]
						ifFalse: [[:each | each isMetaclass not and: [each name beginsWith: aString]]]].
	classes := environment classes select: filter.
```